File: app/src/main/java/se/kicksort/metalarchives/search/BandSearchResultViewHolder.java
package se.kicksort.metalarchives.search;

import com.github.wrdlbrnft.sortedlistadapter.SortedListAdapter;

import io.reactivex.Observable;
import io.reactivex.subjects.PublishSubject;
import se.kicksort.metalarchives.databinding.BandSearchResultBinding;
import se.kicksort.metalarchives.model.BandSearchResult;

public class BandSearchResultViewHolder extends SortedListAdapter.ViewHolder<BandSearchResult> {
    private final BandSearchResultBinding mBinding;
    private PublishSubject<BandSearchResult> clickSubject = PublishSubject.create();

    public BandSearchResultViewHolder(BandSearchResultBinding binding) {
        super(binding.getRoot());
        mBinding = binding;
        mBinding.getRoot().setOnClickListener(view -> clickSubject.onNext(mBinding.getModel()));
    }

    @Override
    public void performBind(BandSearchResult item) {
        mBinding.setModel(item);
    }

    public Observable<BandSearchResult> getSearchResultClicks() {
        return clickSubject;
    }
}

File: app/src/main/java/se/kicksort/metalarchives/search/BandSearchResultViewHolder.kt
package se.kicksort.metalarchives.search

import com.github.wrdlbrnft.sortedlistadapter.SortedListAdapter

import io.reactivex.Observable
import io.reactivex.subjects.PublishSubject
import se.kicksort.metalarchives.databinding.BandSearchResultBinding
import se.kicksort.metalarchives.model.BandSearchResult

class BandSearchResultViewHolder(private val mBinding: BandSearchResultBinding) : SortedListAdapter.ViewHolder<BandSearchResult>(mBinding.root) {
    private val clickSubject = PublishSubject.create<BandSearchResult>()

    val searchResultClicks: Observable<BandSearchResult>
        get() = clickSubject

    init {
        mBinding.root.setOnClickListener { view -> clickSubject.onNext(mBinding.model) }
    }

    public override fun performBind(item: BandSearchResult) {
        mBinding.model = item
    }
}